staging/src/kubevirt.io/api/pool/v1alpha1/types.go

```diff
@@ -170,7 +175,49 @@ type VirtualMachinePoolSelectionPolicy struct {
 	// +optional
 	// +kubebuilder:validation:Enum=Random;DescendingOrder
 	BasePolicy *VirtualMachinePoolBasePolicy `json:"basePolicy,omitempty"`
+
+	// OrderedPolicies is a Ordered list of selection policies.
+	// +optional
+	OrderedPolicies *VirtualMachinePoolOrderedPolicy `json:"orderedPolicies,omitempty"`
+}
+
+// VirtualMachinePoolOrderedPolicy specifies filtering criteria for VM selection.
+// Both LabelSelector and NodeSelectorRequirementMatcher can be specified together.
+// +k8s:openapi-gen=true
+type VirtualMachinePoolOrderedPolicy struct {
+	// LabelSelector is a list of label selector for VMs.
+	// +optional
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+
+	// NodeSelectorRequirementMatcher is a list of node selector requirement for VMs.
+	// +optional
+	NodeSelectorRequirementMatcher *[]k8sv1.NodeSelectorRequirement `json:"nodeSelectorRequirementMatcher,omitempty"`
 }
 
 // +k8s:openapi-gen=true
 type VirtualMachinePoolBasePolicy string
+
+// VirtualMachinePoolUpdateStrategy specifies how the VMPool controller manages updating VMs within a VMPool
+// +k8s:openapi-gen=true
+type VirtualMachinePoolUpdateStrategy struct {
+	// Unmanaged indicates that no automatic update of VMs within a VMPool is performed, by default it is false
+	// +optional
+	Unmanaged *bool `json:"unmanaged,omitempty"`
+
+	// Opportunistic update by updating the VMs in a pool in a non-blocking manner
+	// +optional
+	Opportunistic *bool `json:"opportunistic,omitempty"`
```

Comment:
I think that using `*bool` types in an API is a smell, as it cannot be easily extended in the future. Just like `Proactive` has additional configs, we might want to add configs to these fields as well.

Can you use an empty struct here as well (see references here: [[1](https://github.com/kubevirt/kubevirt/blob/95d6b899d6db35c0ffc4eec73c305d0cd036deb7/staging/src/kubevirt.io/api/core/v1/schema.go#L910)][[2](https://github.com/kubevirt/kubevirt/blob/95d6b899d6db35c0ffc4eec73c305d0cd036deb7/staging/src/kubevirt.io/api/core/v1/schema.go#L128)])?